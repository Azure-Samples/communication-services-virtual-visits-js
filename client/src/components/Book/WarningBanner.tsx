// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { MessageBar, MessageBarType, Text, Link } from '@fluentui/react';
import { BOOKINGS_README_URL } from '../../utils/Constants';

const WarningBanner = (): JSX.Element => {
  return (
    <MessageBar messageBarType={MessageBarType.warning} isMultiline={false} dismissButtonAriaLabel="Close">
      <WarningMessage />
    </MessageBar>
  );
};

// Extracting to a component since MessageBar is not rendering child components in tests
export const WarningMessage = (): JSX.Element => {
  return (
    <>
      <Text>This Booking page is a specimen and will not create appointments or send reminders.</Text>
      <Link href={BOOKINGS_README_URL} target="_blank" underline>
        <Text>Check out the sample readme.</Text>
      </Link>
    </>
  );
};

export default WarningBanner;
