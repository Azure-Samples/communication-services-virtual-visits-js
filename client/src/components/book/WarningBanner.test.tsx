// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import * as renderer from 'react-test-renderer';
import WarningBanner, { WarningMessage } from './WarningBanner';

describe('WarningBanner', () => {
  it('matches snapshot', () => {
    const component = renderer.create(<WarningBanner />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('WarningMessage', () => {
  it('matches snapshot', () => {
    const component = renderer.create(<WarningMessage />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
