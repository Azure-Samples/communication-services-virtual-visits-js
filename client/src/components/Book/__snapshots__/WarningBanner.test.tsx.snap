// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningBanner matches snapshot 1`] = `
<div
  className="ms-MessageBar ms-MessageBar--warning ms-MessageBar-singleline root-109"
>
  <div
    className="ms-MessageBar-content content-110"
  >
    <div
      aria-hidden={true}
      className="ms-MessageBar-icon iconContainer-111"
    >
      <i
        aria-hidden={true}
        className="icon-118"
        data-icon-name="Info"
      />
    </div>
    <div
      aria-live="polite"
      className="ms-MessageBar-text text-113"
      id="MessageBar0"
      role="status"
    >
      <span
        className="ms-MessageBar-innerText innerText-114"
      />
    </div>
  </div>
</div>
`;

exports[`WarningMessage matches snapshot 1`] = `
Array [
  <span
    className="css-119"
  >
    This Booking page is a specimen and will not create appointments or send reminders.
  </span>,
  <a
    className="ms-Link root-120"
    href="https://aka.ms/virtual-appointments-sample-bookings"
    onClick={[Function]}
    target="_blank"
  >
    <span
      className="css-119"
    >
      Check out the sample readme.
    </span>
  </a>,
]
`;
